File: src/main/java/io/bloco/faker/helpers/RandomHelper.java
package io.bloco.faker.helpers;

import java.security.SecureRandom;
import java.util.List;

public class RandomHelper {

    private SecureRandom random;

    public RandomHelper() {
        random = new SecureRandom();
    }

    public <T> T sample(List<T> options) {
        return options.get(random.nextInt(options.size()));
    }

    public String digit() {
        return Integer.toString(random.nextInt(10));
    }

    public boolean randBoolean() {
        return random.nextBoolean();
    }

    public int number(int max) {
        return random.nextInt(max);
    }

    public long number(long max) {
        return Math.abs(random.nextLong()) % max;
    }

    public int numberByLength(int length) {
        return random.nextInt((int) Math.pow(10, length));
    }

    public int range(int min, int max) {
        if (min == max) {
            return min;
        }
        if (max < min) {
            int temp = max;
            max = min;
            min = temp;
        }
        return number(max - min + 1) + min;
    }

    public long range(long min, long max) {
        if (min == max) {
            return min;
        }
        if (max < min) {
            long temp = max;
            max = min;
            min = temp;
        }
        return number(max - min + 1) + min;
    }

    public double range(double min, double max) {
        if (max < min) {
            double temp = max;
            max = min;
            min = temp;
        }
        return random.nextDouble() * (max - min) + min;
    }

    public double randDouble() {
        return random.nextDouble();
    }
}

File: lib/src/main/kotlin/io/bloco/faker/helpers/RandomHelper.kt
package io.bloco.faker.helpers

import java.security.SecureRandom
import kotlin.math.abs
import kotlin.math.pow

class RandomHelper {
    private val random: SecureRandom = SecureRandom()

    fun <T> sample(options: List<T>): T {
        return options[random.nextInt(options.size)]
    }

    fun digit(): String {
        return random.nextInt(10).toString()
    }

    fun randBoolean(): Boolean {
        return random.nextBoolean()
    }

    fun number(max: Int): Int {
        return random.nextInt(max)
    }

    fun number(max: Long): Long {
        return if (max != 0L) abs(random.nextLong()) % max else 0
    }

    fun numberByLength(length: Int): Int {
        return random.nextInt(10.0.pow(length.toDouble()).toInt())
    }

    fun range(min: Int, max: Int): Int {
        var tempMin = min
        var tempMax = max
        if (tempMin == tempMax) {
            return tempMin
        }
        if (tempMax < tempMin) {
            val temp = tempMax
            tempMax = tempMin
            tempMin = temp
        }
        return number(tempMax - tempMin + 1) + tempMin
    }

    fun range(min: Long, max: Long): Long {
        var tempMin = min
        var tempMax = max
        if (tempMin == tempMax) {
            return tempMin
        }
        if (tempMax < tempMin) {
            val temp = tempMax
            tempMax = tempMin
            tempMin = temp
        }
        return number(tempMax - tempMin + 1) + tempMin
    }

    fun range(min: Double, max: Double): Double {
        var tempMin = min
        var tempMax = max
        if (tempMax < tempMin) {
            val temp = tempMax
            tempMax = tempMin
            tempMin = temp
        }
        return random.nextDouble() * (tempMax - tempMin) + tempMin
    }

    fun randDouble(): Double {
        return random.nextDouble()
    }
}